cmd/soroban-cli/src/commands/contract/info/shared.rs

```diff
@@ -54,23 +75,31 @@ pub enum Error {
     Wasm(#[from] wasm::Error),
     #[error("provided wasm hash is invalid {0:?}")]
     InvalidWasmHash(String),
+    #[error("must provide one of --wasm, --wasm-hash, or --contract-id")]
+    MalformedWasmOrWasmHashOrContractId,
     #[error(transparent)]
     Rpc(#[from] soroban_rpc::Error),
     #[error(transparent)]
     Locator(#[from] locator::Error),
 }
 
-pub async fn fetch_wasm(args: &Args) -> Result<Option<Vec<u8>>, Error> {
+pub async fn fetch_wasm(
+    args: &Args,
+    print: &Print,
+) -> Result<(Option<Vec<u8>>, Option<String>, Option<Network>), Error> {
```

Comment:
IMO the signature of this function should be
`Result<(Vec<u8>, Option<String>), Error>`: we can always get the network upstream from network args. 
For SAC I'd say we should error and it can be handled upstream (e.g. in the `contract info` we return hardcoded SAC spec)  -> this simplifies the flow and aligns with function names (fetches WASM)

Reply:
I was trying to do minimal surgery on `contract info` commands. The `Option<Vec<u8>>` is not new here. If we return an error for the case of SAC, we will need to update `contract info interface` and `contract bindings typescript` logic to inspect the error and consider it not-an-error in this specific case. That seems awkward.

As far as omitting the `Network` and requiring caller to re-derive it, I'm fine with that. But I'm not sure it's a big enough cleanup to be worth it?

Reply:
I think it's preferred to pass the Network back. Deriving it multiple times could lead to divergence due to a bug, and then confusing output. I left a comment about the return value of the function in another thread below.

Reply:
Alternatively pass the network in.